simple/manifest.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package simple
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+// Error used when lookup path does not match
+var (
+	ErrNotFound  = errors.New("not found")
+	ErrEmptyPath = errors.New("empty path")
+)
+
+// Manifest is a JSON representation of a manifest.
+// It stores manifest entries in a map based on string keys.
+type Manifest struct {
+	Entries map[string]*Entry `json:"entries,omitempty"`
+
+	mu sync.RWMutex // mutex for accessing the entries map
+}
```

Comment:
I know that it is currently the same in bee jsonmanifest package, but exporting a map that is protected by unexported mutex opens a possibility of data race by the user of the package. I would prioritize correctness of the exported go package api instead a simpler implementation required for json encoding. Given that the same issue exists in the bee, this can be solved later.

Reply:
Yeah, those fields (in `Entry` also) must be exported due to JSON support. Not sure if mutex is needed anyway, especially since I still have not seen the need for concurrent access, as well as `mantaray` implementation is also not safe.

Anyhow, I can fix this also.

Reply:
That is a good point. Thanks for addressing this. Concurrent access protection is there, but if we do not need it, it can be also removed. Maybe in some followup PR, if needed.